File: vncccd-sdk-ios/Sources/VNCCCDSDK/PersonalInfo.swift
import Foundation

public struct PersonalInfo: Sendable, Codable, Equatable {
    public let fullName: String?
    public let idNumber: String?
    public let dateOfBirth: String?
    public let gender: String?
    public let nationality: String?
    public let ethnicity: String?
    public let religion: String?
    public let placeOfOrigin: String?
    public let placeOfResidence: String?
    public let personalIdentification: String?
    public let dateOfIssue: String?
    public let dateOfExpiry: String?
    public let fatherName: String?
    public let motherName: String?
    public let spouseName: String?
    public let oldIdNumber: String?

    public init(
        fullName: String? = nil,
        idNumber: String? = nil,
        dateOfBirth: String? = nil,
        gender: String? = nil,
        nationality: String? = nil,
        ethnicity: String? = nil,
        religion: String? = nil,
        placeOfOrigin: String? = nil,
        placeOfResidence: String? = nil,
        personalIdentification: String? = nil,
        dateOfIssue: String? = nil,
        dateOfExpiry: String? = nil,
        fatherName: String? = nil,
        motherName: String? = nil,
        spouseName: String? = nil,
        oldIdNumber: String? = nil
    ) {
        self.fullName = fullName
        self.idNumber = idNumber
        self.dateOfBirth = dateOfBirth
        self.gender = gender
        self.nationality = nationality
        self.ethnicity = ethnicity
        self.religion = religion
        self.placeOfOrigin = placeOfOrigin
        self.placeOfResidence = placeOfResidence
        self.personalIdentification = personalIdentification
        self.dateOfIssue = dateOfIssue
        self.dateOfExpiry = dateOfExpiry
        self.fatherName = fatherName
        self.motherName = motherName
        self.spouseName = spouseName
        self.oldIdNumber = oldIdNumber
    }
}

File: vncccd-sdk/src/main/kotlin/com/vncccd/sdk/models/PersonalInfo.kt
package com.vncccd.sdk.models

import java.io.Serializable

/**
 * Thông tin cá nhân đọc từ DG13 của chip CCCD.
 * DG13 chứa thông tin đặc thù Việt Nam mà DG1 (MRZ) không thể chứa hết.
 */
data class PersonalInfo(
    /** Họ và tên đầy đủ (tiếng Việt có dấu) */
    val fullName: String? = null,

    /** Số CCCD đầy đủ (12 chữ số) */
    val idNumber: String? = null,

    /** Ngày sinh (dd/MM/yyyy) */
    val dateOfBirth: String? = null,

    /** Giới tính (Nam/Nữ) */
    val gender: String? = null,

    /** Quốc tịch */
    val nationality: String? = null,

    /** Dân tộc */
    val ethnicity: String? = null,

    /** Tôn giáo */
    val religion: String? = null,

    /** Quê quán */
    val placeOfOrigin: String? = null,

    /** Nơi thường trú */
    val placeOfResidence: String? = null,

    /** Đặc điểm nhận dạng */
    val personalIdentification: String? = null,

    /** Ngày cấp (dd/MM/yyyy) */
    val dateOfIssue: String? = null,

    /** Ngày hết hạn (dd/MM/yyyy) */
    val dateOfExpiry: String? = null,

    /** Họ tên cha */
    val fatherName: String? = null,

    /** Họ tên mẹ */
    val motherName: String? = null,

    /** Họ tên vợ/chồng */
    val spouseName: String? = null,

    /** Số CCCD cũ / CMND */
    val oldIdNumber: String? = null
) : Serializable {
    companion object {
        private const val serialVersionUID = 1L
    }
}
